src/API/Google/MerchantReport.php

```diff
@@ -44,12 +59,114 @@ class MerchantReport implements OptionsAwareInterface {
 	/**
 	 * Merchant Report constructor.
 	 *
-	 * @param ShoppingContent $service
-	 * @param ProductHelper   $product_helper
+	 * @param ShoppingContent   $service
+	 * @param ProductHelper     $product_helper
+	 * @param ProductRepository $product_repository
 	 */
-	public function __construct( ShoppingContent $service, ProductHelper $product_helper ) {
-		$this->service        = $service;
-		$this->product_helper = $product_helper;
+	public function __construct( ShoppingContent $service, ProductHelper $product_helper, ProductRepository $product_repository ) {
+		$this->service            = $service;
+		$this->product_helper     = $product_helper;
+		$this->product_repository = $product_repository;
+	}
+
+	/**
+	 * Get product statuses.
+	 *
+	 * @return array List of product statuses.
+	 * @throws GoogleException If the search call fails.
+	 */
+	public function get_product_statuses(): array {
+		$sync_ready_products_ids = $this->product_repository->find_sync_ready_products()->get_product_ids();
+
+		if ( count( $sync_ready_products_ids ) === 0 ) {
+			return $this->calculate_statuses( [] );
+		}
+
+		$offer_ids = array_map(
+			function ( $item ) {
+				return WCProductAdapter::get_offer_id( $item );
+			},
+			$sync_ready_products_ids
+		);
+
+		$next_page_token = null;
+		$results         = [];
+
+		do {
+
+			$query = new MerchantProductViewReportQuery(
+				[
+					'ids'       => $offer_ids,
+					'next_page' => $next_page_token,
+				]
+			);
+
+			/** @var SearchResponse $response  */
+			$response = $query
+			->set_client( $this->service, $this->options->get_merchant_id() )
+			->get_results();
+
+			if ( $response->count() ) {
+				$results = [ ...$results, ...$response->getResults() ];
+			}
+
+			$next_page_token = $response->getNextPageToken();
+
+		} while ( $next_page_token );
+
+		$statistics               = $this->calculate_statuses( $results );
+		$statistics['not_synced'] = count( $sync_ready_products_ids ) - count( $results );
+		return $statistics;
+	}
+
+	/**
+	 * Calculate statistics for a list of report rows.
+	 *
+	 * @param ReportRow[] $rows List of report rows.
+	 *
+	 * @return array List of statistics.
+	 */
+	public function calculate_statuses( array $rows ): array {
+		$statistics = [
+			'active'              => 0,
+			'not_synced'          => 0,
+			MCStatus::EXPIRING    => 0,
+			MCStatus::PENDING     => 0,
+			MCStatus::DISAPPROVED => 0,
+		];
+
+		/** @var $row ReportRow  */
+		foreach ( $rows as $row ) {
+			/** @var ProductView $product_view  */
+			$product_view    = $row->getProductView();
+			$experation_date = $product_view->getExpirationDate();
```

Comment:
Just a typo here:
```suggestion
			$expiration_date = $product_view->getExpirationDate();
```
Same with all the later references where the variable is used.